### PR TITLE
Orca: Deprecate `scheduler_step_freq` in torch estimator

### PR DESCRIPTION
**Pull request by leonardozcm**
## Description

In this PR we deprecate `scheduler_step_freq` in torch estimator for our users can step scheduler everywhere they prefer with maincallback, and `scheduler_step_freq` may seem a bit redundant.

we will provide a correspondding usage example in hook's document later.

### CODE REVIEW

File: docs/docs/Orca/orca-pytorch-estimator.md [hunk 1]
```diff
@@ -43,11 +42,10 @@ Estimator.from_torch(*,
 * `optimizer`: Orca optimizer or PyTorch optimizer if `backend="bigdl"`, PyTorch optimizer creator if `backend="horovod" or "ray"`
 * `loss`: PyTorch loss if `backend="bigdl"`, PyTorch loss creator if `backend="horovod" or "ray"`
 * `metrics`: Orca validation methods for evaluate.
-* `scheduler_creator`: parameter for `horovod` and `ray` backends. a learning rate scheduler wrapping the optimizer. You will need to set ``scheduler_step_freq="epoch"`` for the scheduler to be incremented correctly.
+* `scheduler_creator`: parameter for `horovod` and `ray` backends. a learning rate scheduler wrapping the optimizer. By default a scheduler will increment automatically every epoch.
```
Comment (hkvision) [inline docs/docs/Orca/orca-pytorch-estimator.md:45]: spark, ray and horovod backends?

File: python/orca/example/learn/horovod/pytorch_estimator.py [hunk 2]
```diff
@@ -51,8 +51,8 @@ def optimizer_creator(model, config):
 
 def scheduler_creator(optimizer, config):
     """Returns a learning rate scheduler wrapping the optimizer.
-    You will need to set ``TorchTrainer(scheduler_step_freq="epoch")``
-    for the scheduler to be incremented correctly.
+    By default a scheduler will increment automatically every epoch,
```
Comment (hkvision) [inline python/orca/example/learn/horovod/pytorch_estimator.py:54]: increment -> take effect

File: python/orca/example/learn/horovod/pytorch_estimator.py [hunk 3]
```diff
@@ -51,8 +51,8 @@ def optimizer_creator(model, config):
...
+    and you can update the scheduler at the right time through the hooks.
```
Comment (hkvision) [inline python/orca/example/learn/horovod/pytorch_estimator.py:55]: you can customize the behavior of the scheduler through the MainHook?

File: python/orca/src/bigdl/orca/learn/pytorch/estimator.py [hunk 4]
```diff
@@ -74,12 +73,8 @@ def from_torch(*,
                Default: 1.
         :param scheduler_creator: A scheduler creator function that has two parameters "optimizer"
                and "config" and returns a PyTorch learning rate scheduler wrapping the optimizer.
-               Note that if you specify this parameter, you need to take care of the argument
-               scheduler_step_freq accordingly as well.
+               By default a scheduler will increment automatically every epoch.
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/learn/pytorch/estimator.py:76]: same as above

**Review by hkvision — APPROVED** [review]

File: python/orca/example/learn/horovod/pytorch_estimator.py (hunk 3 above)
Comment (leonardozcm) [reply python/orca/example/learn/horovod/pytorch_estimator.py:55]: > you can customize the behavior of the scheduler through the MainHook?

Yes, by overriding `on_lr_adjust` and other hooks.For example if you want to step by iteration:
```python
def on_lr_adjust(self, runner):
      pass

def after_train_iter(self, runner):
      runner.scheduler.step()
```

Comment (hkvision) [reply python/orca/example/learn/horovod/pytorch_estimator.py:55]: I mean modify the doc... not explain to me...😂

Comment (leonardozcm) [reply python/orca/example/learn/horovod/pytorch_estimator.py:55]: > I mean modify the doc... not explain to me...😂

All right😂then as I mentioned in the description I will add this later, together with some other typical usage cases.
